src/flow/src/adapter.rs

```diff
@@ -345,16 +343,32 @@ impl FlownodeManager {
                             .names
                             .get(idx)
                             .cloned()
+                            .flatten()
                             .unwrap_or(format!("Col_{}", idx));
-                        ColumnSchema::new(name, typ.scalar_type, typ.nullable)
+                        let ret = ColumnSchema::new(name, typ.scalar_type, typ.nullable);
+                        if schema.typ().time_index == Some(idx) {
+                            ret.with_time_index(true)
+                        } else {
+                            ret
+                        }
                     })
                     .collect_vec();
 
                 let mut with_ts = wout_ts.clone();
                 with_ts.push(update_at);
-                with_ts.push(ts_col);
 
-                (primary_keys, with_ts, true)
+                // if no time index, add one
+                if schema.typ().time_index.is_none() {
+                    let ts_col = ColumnSchema::new(
+                        AUTO_CREATED_PLACEHOLDER_TS_COL,
+                        ConcreteDataType::timestamp_millisecond_datatype(),
+                        true,
+                    )
+                    .with_time_index(true);
+                    with_ts.push(ts_col);
+                }
+
+                (primary_keys, with_ts, schema.typ().time_index.is_none())
```

Comment:
Here `schema.typ().time_index.is_none()` is always true, which cannot indicate that the ts column is a plaeholder

Reply:
> Here `schema.typ().time_index.is_none()` is always true, which cannot indicate that the ts column is a plaeholder

Added a `find_time_index_in_group_exprs` function to assign time index now, so there is cases where `time_index.is_some()`